src/features/api/apiSlice.js

```diff
@@ -0,0 +1,17 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export const apiSlice = createApi({
+  reducerPath: "api",
+  baseQuery: fetchBaseQuery({ baseUrl: "https://dummyjson.com" }),
+  endpoints: (builder) => ({
+    getProducts: builder.query({ query: () => "/products?limit=10" }),
+    getCategories: builder.query({ query: () => "/products/categories" }),
+    getSingleProduct: builder.query({ query: (id) => `products/${id}` }),
```

Comment:
is often just named 'getProduct' (singular) implying that the identifier is needed as argument

Reply:
Aright!